public/app/features/explore/AddToDashboard/addToDashboard.ts

```diff
@@ -25,13 +24,25 @@ const createDashboard = (
   return getDashboardSrv().saveDashboard({ dashboard, folderId }, { showErrorAlert: false, showSuccessAlert: false });
 };
 
-export const addToDashboard = async (data: SaveToNewDashboardDTO, options: SaveOptions): Promise<string> => {
-  const res = await createDashboard(
-    data.dashboardName,
-    data.folderId,
-    options.queries,
-    options.datasource,
-    options.panel
-  );
-  return res.data.url;
+const addPanelToDashboard = async (uid: string, queries: DataQuery[], datasource: DataSourceRef, panel: string) => {
+  const dashboardData = await backendSrv.getDashboardByUid(uid);
+
+  const dashboard = new DashboardModel(dashboardData.dashboard, dashboardData.meta);
+
+  dashboard.addPanel({ targets: queries, type: panel, title: 'New Panel', datasource });
```

Comment:
this will just add it to the top of the dashboard (since no gridPos is added), breaking the layout.